wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/ConversationGuestOptionsViewModel.swift

```diff
@@ -232,26 +244,37 @@ final class ConversationGuestOptionsViewModel {
             self?.showLoadingCell = true
         }
 
-        configuration.createConversationLink { [weak self] result in
-            guard let `self` = self else { return }
+        createSecureGuestLinkUseCase.invoke(conversation: conversation, password: nil) { result in
             switch result {
-            case .success(let link): self.link = link
-            case .failure(let error): self.delegate?.viewModel(self, didReceiveError: error)
+            case .success(let link):
+                self.link = link
+            case .failure(let error):
+                self.delegate?.viewModel(self, didReceiveError: error)
             }
 
             item.cancel()
             self.showLoadingCell = false
         }
+
     }
 
     /// Starts the Guest Link Creation Flow
     /// - Parameter view: the source view which triggers create action
     func startGuestLinkCreationFlow(from view: UIView? = nil) {
         if isGuestLinkWithPasswordAvailable {
-            delegate?.viewModel(self, sourceView: view, presentGuestLinkTypeSelection: { [weak self] guestLinkType in
+            delegate?.viewModel(self,
+                                sourceView: view,
+                                presentGuestLinkTypeSelection: { [weak self] guestLinkType in
                 guard let `self` = self else { return }
                 switch guestLinkType {
-                case .secure: break
+                case .secure:
+                    delegate?.viewModel(
+                        self,
+                        presentCreateSecureGuestLink: CreateSecureGuestLinkViewController(
+                            conversationSecureGuestLinkUseCase: createSecureGuestLinkUseCase,
+                            conversation: conversation).wrapInNavigationController(),
```

Comment:
It would be easier to read if we make a let out of it.
```swift
let viewController = CreateSecureGuestLinkViewController(
    conversationSecureGuestLinkUseCase: createSecureGuestLinkUseCase,
    conversation: conversation
)
let navigationController = viewController.wrapInNavigationController()
```

Reply:
Done!